### DISCUSSION TITLE
Is it possible to automate a pg_dump to be done before a delete command?

### DISCUSSION THREAD

**Discussion by CrawliestCotter** [netbox-community/netbox#17519] [General]
I'm looking for a way to create a PSQL snapshot/dump (pg_dump) when someone is trying to delete items. What I'm trying to do is hook a prompt to the deletion confirmation asking if you want to create a snapshot first.

It'd also be really slick if there could be a db restore integration in the changelog to restore to an available snapshot made during a change.

For now, I'll just make it protocol to do a pg_dump before any deletions via the cli.

**Comment by ghost** [comment]
https://github.com/netboxlabs/netbox-branching ??

**Reply by CrawliestCotter** [reply]
I'll take a look into that. Thanks!

**Reply by CrawliestCotter** [reply]
This is close, but it requires a lot of forthought to on the operator side to create a branch before they make changes. I'm also having some problems with it not being able to revert branch merges when deleting devices with interface connections. If reversion fails, there's no way to rebase, so you're stuck manually entering the deleted items again.

Very interesting and exciting project though. They've been made some great progress for it being such a new project, and I can see it really becoming a powerful tool.

**Reply by ghost** [reply]
If the branching has problems, head over to the branch git and do a FR or bug report.

**Reply by CrawliestCotter** [reply]
I did so, but the branching system is still not exactly what I'm looking for. Really it looks like the scripting and event system might be able to do what I'm looking for. might do some testing down that line to see if I can get some sort of snap-shot system working.
